Source/WebKit/WebProcess/WebPage/Cocoa/TextAnimationController.mm

```diff
@@ -182,9 +182,9 @@
 {
     auto location = previousRange.location + previousRange.length;
     auto length = totalRange.length - previousRange.length;
+
     if (totalRange.length < previousRange.length) {
-        ASSERT_NOT_REACHED();
-        return WebCore::CharacterRange { 0, 0 };
+        return totalRange;
```

Comment:
in what case does this happen? Also, this should probably move above the

```
auto length = totalRange.length - previousRange.length
```

line so that `length` isn't underflowed (it doesn't actually matter since it's not beeng used in this case anyways, but it's still better for correctness)

Reply:
"The previous range was adjusted to the the actual range of the text before it was stored
but we were comparing that against the full range of text, which was incorrect
and also if the replaced text was shorter"